### PR TITLE
🗑️ cleanup deprecated idf 4.3 related code

### PR DESCRIPTION
**Pull request by Vollbrecht**

### CODE REVIEW

File: build/native/cargo_driver/config.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ use crate::config::EspIdfSys;
 pub const ESP_IDF_VERSION_VAR: &str = "ESP_IDF_VERSION";
 pub const ESP_IDF_REPOSITORY_VAR: &str = "ESP_IDF_REPOSITORY";
 
-pub const DEFAULT_ESP_IDF_VERSION: &str = "v4.4.6";
+pub const DEFAULT_ESP_IDF_VERSION: &str = "v5.2.1";
```
Comment (MabezDev) [inline build/native/cargo_driver/config.rs:19]: Nice :+1:

File: src/patches.rs [hunk 2]
```diff
@@ -3,30 +3,22 @@
...
-mod pthread_rwlock;
 
 #[allow(dead_code)]
 pub struct PatchesRef(
     *mut core::ffi::c_void,
     *mut core::ffi::c_void,
-    *mut core::ffi::c_void,
 );
 
 /// A hack to make sure that the rwlock implementation is linked to the final executable
```
Comment (MabezDev) [inline src/patches.rs:13]: The doc comment needs updating too

Comment (MabezDev) [reply src/patches.rs:13]: Oops the diff messed up here, but I'm referring to the comment below

Comment (Vollbrecht) [reply src/patches.rs]: updated it, though i am not to sure myself what __LSTAT_INTERNAL_REFERENCE is, though the other thing we link is more clear about the atexit ;D

Comment (Vollbrecht) [reply src/patches.rs]: so i just call it together "certain symbols" ;D

**Review by ivmarkov — CHANGES_REQUESTED** [review]

File: src/patches.rs [hunk 3]
```diff
@@ -28,5 +20,12 @@ pub fn link_patches() -> PatchesRef {
...
     let atexit = atexit::link_patches();
 
-    PatchesRef(rwlock, lstat, atexit)
+    PatchesRef(lstat, atexit)
 }
+
+#[cfg(esp_idf_version_major = "4")]
+#[deprecated(
+    since = "0.35.0",
+    note = "ESP_IDF version 4 is now considerd deprecated.\nPlease upgrade your project to ESP_IDF 5 or later.\nFuture releases will remove parts that are v4 related"
```
[the remark this answers is not in the document]
Comment (Vollbrecht) [reply src/patches.rs]: fixed

**Pull request merged**